app/pages/lab-pages-editor/components/WorkflowSettingsPage/components/AssociatedTutorial.jsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useState } from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+
+const ARBITRARY_PAGE_SIZE = 100;  // If a project has more than this number of tutorials, then we need a better solution.
+
+export default function AssociatedTutorial({ project, workflow }) {
+  const [ linkedTutorial, setLinkedTutorial ] = useState(null); 
+  const [apiData, setApiData] = useState({
+    tutorials: null,
+    status: 'ready'
+  });
+
+  useEffect(() => {
+    async function fetchTutorials() {
+      try {
+        if (!workflow || !project) throw new Error('No workflow and/or no project.');
+
+        setApiData({
+          tutorials: null,
+          status: 'fetching'
+        });
+
+        // First we need to fetch all tutorials associated with the project.
+        const tutorialsLinkedToProject = await apiClient.type('tutorials').get({ project_id: project.id, page_size: ARBITRARY_PAGE_SIZE });
+        if (!tutorialsLinkedToProject) throw new Error('No tutorials');
```

Comment:
If a project and/or workflow chooses not to have a tutorial should it raise an error? What happens to a new project that doesn't have any tutorials associated with it? Part of the reason I ask this is that I personally only throw errors when the code should not continue execution on error... And I don't get the sense that's the case here.

Reply:
❗ Oh, that's a good point! Thank you for catching this, I'll need to test with a project with no tutorials.

Reply:
OK, I've finally got clarification on this. 👌 

If a project has **no tutorials,** then `apiClient.type('tutorials').get(...)` returns `[]`, which is valid. If apiClient returns _undefined_ however, then that's a fetch issue.

I'll need to run more thorough tests and add more descriptive errors (visible to users) though. I'll need a separate PR just examining the ways things can break (e.g. network errors) and craft appropriate error messages.